rust/src/lib.rs

```diff
@@ -1201,6 +1239,43 @@ pub extern "C" fn zcashlc_get_sent_memo_as_utf8(
     unwrap_exc_or_null(res)
 }
 
+/// Returns the memo for a sent note, by copying the corresponding bytes to the received
+/// pointer in `memo_bytes_ret`.
+///
+/// The note is identified by its row index in the `sent_notes` table within the data
+/// database.
+///
+/// # Safety
+///
+/// - `db_data` must be non-null and valid for reads for `db_data_len` bytes, and it must have an
+///   alignment of `1`. Its contents must be a utf-8 string representing a valid system path.
+/// - The memory referenced by `db_data` must not be mutated for the duration of the function call.
+/// - The total size `db_data_len` must be no larger than `isize::MAX`. See the safety
+///   documentation of pointer::offset.
+/// - `memo_bytes_ret` must be non-null and must point to an allocated 512-byte region of memory.
+#[no_mangle]
+pub extern "C" fn zcashlc_get_sent_memo(
+    db_data: *const u8,
+    db_data_len: usize,
+    id_note: i64,
+    memo_bytes_ret: *mut [u8; 512],
+    network_id: u32,
+) -> bool {
```

Comment:
This only differs from `zcashlc_get_received_memo` by the enum variant of the `NoteId`. Perhaps factor out the near-duplicate code?